internal/zinc-benchmarks/src/test/scala/xsbt/AnalysisFormatBenchmark.scala

```diff
@@ -0,0 +1,176 @@
+package xsbt
+
+import java.io.File
+import java.util.concurrent.TimeUnit
+import scala.collection.mutable
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+import sbt.internal.inc.consistent._
+import sbt.internal.inc.{ Analysis, FileAnalysisStore }
+import sbt.io.IO
+import xsbti.compile.analysis.ReadWriteMappers
+import xsbti.compile.{ AnalysisContents, AnalysisStore }
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+class AnalysisFormatBenchmark {
```

Comment:
Can we integrate it as part of CI check, similar to https://github.com/sbt/zinc/pull/1323?

This doesn't have to go into this PR, we can do this in a separate PR.